web/src/services/api/pluginsApi.ts

```diff
@@ -31,47 +39,186 @@ export type Plugin = {
-                };
-              })
-              .filter(p => !!p) as Plugin[])
-          : undefined,
-      [data?.node],
-    );
-
-    return { plugins, ...rest };
-  }, []);
+  const [, setNotification] = useNotification();
+  const lang = useLang();
+  const t = useT();
+
+  const usePluginsQuery = useCallback(
+    (sceneId?: string) => {
+      const { data, ...rest } = useQuery(GET_SCENE, {
+        variables: { sceneId: sceneId ?? "", lang },
+        skip: !sceneId,
+      });
+
+      const plugins: Plugin[] | undefined = useMemo(
+        () =>
+          data?.node?.__typename === "Scene"
+            ? (data.node.plugins
+                .map(p => {
+                  if (!p.plugin) return;
+
+                  return {
+                    id: p.plugin.id,
+                    name: p.plugin.name,
+                    extensions: p.plugin.extensions.map(e => {
+                      return {
+                        pluginId: p.plugin?.id,
+                        ...e,
+                      };
+                    }),
+                    property: p.property
+                      ? {
+                          id: p.property.id,
+                          items: p.property.items,
+                          schema: p.property.schema,
+                        }
+                      : {},
+                  };
+                })
+                .filter(p => !!p) as Plugin[])
+            : undefined,
+        [data?.node],
+      );
+
+      return { plugins, ...rest };
+    },
+    [lang],
+  );
+
+  const [installPluginMutation] = useMutation(INSTALL_PLUGIN, {
+    refetchQueries: ["GET_SCENE"],
+  });
+  const [upgradePluginMutation] = useMutation(UPGRADE_PLUGIN, {
+    refetchQueries: ["GET_SCENE"],
+  });
+
+  const useInstallPlugin = useCallback(
+    async (sceneId: string, pluginId: string) => {
+      if (!sceneId || !pluginId) return;
+
+      const { errors } = await installPluginMutation({
+        variables: { sceneId, pluginId },
+      });
+
+      if (errors) {
+        setNotification({
+          type: "error",
+          text: t("Failed to install plugin."),
+        });
+      } else {
+        setNotification({
+          type: "success",
+          text: t("Successfully installed plugin!"),
+        });
+      }
+    },
+    [installPluginMutation, t, setNotification],
+  );
+
+  const useUpgradePlugin = useCallback(
+    async (sceneId: string, pluginId: string, oldPluginId: string) => {
+      if (!sceneId || !pluginId || !oldPluginId) return;
+
+      const { errors } = await upgradePluginMutation({
+        variables: {
+          sceneId,
+          pluginId: oldPluginId,
+          toPluginId: pluginId,
+        },
+      });
+
+      if (errors) {
+        setNotification({
+          type: "error",
+          text: t("Failed to install plugin."),
+        });
+      } else {
+        setNotification({
+          type: "success",
+          text: t("Successfully installed plugin!"),
```

Comment:
`... upgrade plugin`, right? Not install

Reply:
Updated.